test/e2e/nodenetworkpolicy_test.go

```diff
@@ -891,3 +896,116 @@ func testNodeACNPNestedIPBlockClusterGroupCreateAndUpdate(t *testing.T) {
+	toBeTruncated := "thisPartShouldBeTruncated"
+	logLabel := randomLogLabel + toBeTruncated // Append the suffix to the logLabel.
+	builder := &ClusterNetworkPolicySpecBuilder{}
+	builder = builder.SetName("acnp-drop-x-to-y-egress").
+		SetPriority(1.0).
+		SetAppliedToGroup([]ACNPAppliedToSpec{{NodeSelector: map[string]string{labelNodeHostname: nodes["x"]}}})
+	builder.AddEgress(ProtocolTCP, &p80, nil, nil, nil, nil, nil, nil, nil, nil, map[string]string{labelNodeHostname: nodes["y"]}, nil,
+		nil, nil, nil, nil, nil, crdv1beta1.RuleActionDrop, "", "", nil)
+	builder.AddEgressLogging(logLabel)
+
+	acnp, err := k8sUtils.CreateOrUpdateACNP(builder.Get())
+	require.NoError(t, err)
+	require.NoError(t, data.waitForACNPRealized(t, acnp.Name, policyRealizedTimeout))
+
+	antreaPodName, err := data.getAntreaPodOnNode(nodes["x"])
+	require.NoError(t, err)
+
+	expectedLogPrefix := fmt.Sprintf("Antrea:O:Drop:%s:", randomLogLabel)
+
+	// Before generating some traffic, there should be no corresponding kernel log on hostNetwork Pod x/a, and the kernel
+	// logs should not contain the generated random logLabel.
+	stdout, stderr, err := data.RunCommandFromPod(antreaNamespace, antreaPodName, agentContainerName, []string{"sh", "-c", "dmesg | tail -n 200"})
+	if err != nil || stderr != "" {
+		t.Fatalf("Got error: %v, %v", err, stderr)
+	}
+	require.NotContains(t, stdout, expectedLogPrefix)
+
+	// Generate some traffic that will be dropped by acnp-drop-x-to-y-egress. In this test, the target IP is the Antrea
+	// gateway IP of Node y.
+	_, err = k8sUtils.Probe(getNS("x"), "a", getNS("y"), "a", p80, ProtocolTCP, nil, nil)
+	require.NoError(t, err)
+
+	ipv4Enabled, ipv6Enabled := isIPv4Enabled(), isIPv6Enabled()
+	var srcIP4, dstIP4, srcIP6, dstIP6 string
+	nodeInfo := getNodeByName(nodes["x"])
+	require.NotNil(t, nodeInfo)
+	srcIP4, srcIP6 = nodeGatewayIPs(nodeInfo.idx)
+	nodeInfo = getNodeByName(nodes["y"])
+	require.NotNil(t, nodeInfo)
+	dstIP4, dstIP6 = nodeGatewayIPs(nodeInfo.idx)
+	if ipv4Enabled {
+		require.NotEqual(t, "", srcIP4)
+		require.NotEqual(t, "", dstIP4)
+	}
+	if ipv6Enabled {
+		require.NotEqual(t, "", srcIP6)
+		srcIP6 = expandIPv6(srcIP6)
+		require.NotEqual(t, "", dstIP6)
+		dstIP6 = expandIPv6(dstIP6)
+	}
+
+	re4 := regexp.MustCompile(fmt.Sprintf(`%s.*SRC=%s DST=%s`, expectedLogPrefix, srcIP4, dstIP4))
+	re6 := regexp.MustCompile(fmt.Sprintf(`%s.*SRC=%s DST=%s`, expectedLogPrefix, srcIP6, dstIP6))
+
+	assert.EventuallyWithT(t, func(tc *assert.CollectT) {
+		// After generating some traffic, there should be corresponding kernel logs on hostNetwork Pod x/a, and the kernel
+		// logs should contain the generated random logLabel and not contain the part that should be truncated.
+		stdout, stderr, err = data.RunCommandFromPod(antreaNamespace, antreaPodName, agentContainerName, []string{"sh", "-c", "dmesg | tail -n 200"})
+		if err != nil || stderr != "" {
+			t.Fatalf("Got error: %v, %v", err, stderr)
+		}
+		logs := strings.Split(stdout, "\n")
```

Comment:
nit: I think looping over the log lines is a bit of an overkill
I would just call match on stdout directly to make the test smaller

Reply:
Done